src/server.rs

```diff
@@ -127,29 +131,25 @@ macro_rules! messages {
                 });
             }
 
-            let id: Option<usize> = match ls_command.get("id") {
-                Some(v) => {
-                    if v.is_u64() {
-                        Some(v.as_u64().unwrap() as usize)
-                    } else if v.is_string() {
-                        match usize::from_str_radix(v.as_str().unwrap(), 10) {
-                            Ok(v) => Some(v),
-                            Err(_) => None
-                        }
-                    } else {
-                        None
-                    }
-                },
-                None => None
+            let id = ls_command.get("id").expect("no id").to_owned();
+            let id: jsonrpc::Id = serde_json::from_value(id).expect("Can't extract id");
```

Comment:
We should continue to be careful not to panic here. If we don't need an id, then we should not panic on an invalid id.